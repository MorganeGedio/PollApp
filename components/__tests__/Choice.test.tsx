import React from "react";
import renderer from "react-test-renderer";
import ChoiceItem from "../Choice";

describe("ChoiceItem", () => {
  const testProps = {
    title: "Question Test",
    votes: 12,
    showVote: false,
    disabled: false,
    onPress: jest.fn(),
  };

  test("Choice component renders correctly", () => {
    // render a component with the new renderer
    const component = renderer.create(<ChoiceItem {...testProps} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
