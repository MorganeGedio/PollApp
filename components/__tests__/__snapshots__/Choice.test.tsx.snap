// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChoiceItem Choice component renders correctly 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffe8df",
        "borderRadius": 10,
        "height": 70,
        "justifyContent": "center",
        "marginLeft": 30,
        "marginRight": 30,
        "marginTop": 30,
        "opacity": 1,
        "padding": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.23,
        "shadowRadius": 2.62,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontFamily": "nunito-regular",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Question Test
      </Text>
    </View>
  </View>
</View>
`;
